From 30aa48c616405d56768769f722a7bd09936648d5 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Wed, 17 Jan 2018 17:49:19 -0800
Subject: [PATCH 0750/1239] configs: t93: move to Kconfig defines and cleanup

Define all required entries through menuconfig and remove
stale items in thunderx_93xx.h config file.

Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 configs/t93_defconfig           |  53 +++---
 include/configs/thunderx_93xx.h | 277 +++++---------------------------
 2 files changed, 69 insertions(+), 261 deletions(-)

diff --git a/configs/t93_defconfig b/configs/t93_defconfig
index f3e02195a2..9f0ab2e22c 100644
--- a/configs/t93_defconfig
+++ b/configs/t93_defconfig
@@ -1,39 +1,48 @@
 CONFIG_ARM=y
 # CONFIG_ARM64_SUPPORT_AARCH32 is not set
 CONFIG_ARCH_THUNDERX=y
+CONFIG_SYS_TEXT_BASE=0x04000000
+CONFIG_IDENT_STRING=" for Cavium OcteonTX2 CN93XX ARM V8 Core"
+# CONFIG_PSCI_RESET is not set
 CONFIG_TARGET_THUNDERX_93XX=y
-CONFIG_DEFAULT_DEVICE_TREE="thunder-93xx"
 CONFIG_DEBUG_UART=y
+CONFIG_AHCI=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_OF_BOARD_SETUP=y
-CONFIG_SYS_EXTRA_OPTIONS="ARM64"
-CONFIG_BOOTDELAY=5
+CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="console=ttyAMA0,115200n8 earlycon=pl011,0x87e028000000 debug maxcpus=24 rootwait rw root=/dev/sda2 coherent_pool=16M"
+CONFIG_VERSION_VARIABLE=y
 # CONFIG_DISPLAY_CPUINFO is not set
 # CONFIG_DISPLAY_BOARDINFO is not set
+CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_HUSH_PARSER=y
-# CONFIG_CMD_BOOTEFI_HELLO_COMPILE is not set
+CONFIG_SYS_PROMPT="Cavium> "
+# CONFIG_AUTOBOOT is not set
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
 # CONFIG_CMD_FLASH is not set
+# CONFIG_CMD_FPGA is not set
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
 CONFIG_CMD_PART=y
+CONFIG_CMD_PCI=y
 CONFIG_CMD_SF=y
-CONFIG_CMD_I2C=y
 CONFIG_CMD_USB=y
-CONFIG_CMD_NVME=y
-# CONFIG_CMD_FPGA is not set
-CONFIG_CMD_GPIO=y
 CONFIG_CMD_TFTPPUT=y
 CONFIG_CMD_TFTPSRV=y
 CONFIG_CMD_RARP=y
 CONFIG_CMD_DHCP=y
+CONFIG_CMD_PXE=y
+CONFIG_CMD_MII=y
 CONFIG_CMD_PING=y
 CONFIG_CMD_CDP=y
 CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
 CONFIG_CMD_BOOTIMGUP=y
+CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_CMD_FAT=y
@@ -41,42 +50,42 @@ CONFIG_CMD_FS_GENERIC=y
 CONFIG_EFI_PARTITION=y
 CONFIG_PARTITION_TYPE_GUID=y
 CONFIG_OF_BOARD=y
+CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_SCSI_AHCI=y
 CONFIG_AHCI_PCI=y
 CONFIG_BLOCK_CACHE=y
-CONFIG_NVME=y
+CONFIG_DM_GPIO=y
+CONFIG_DM_I2C=y
 CONFIG_MISC=y
 # CONFIG_MMC is not set
 CONFIG_DM_MMC=y
 CONFIG_MMC_CAVIUM=y
+CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_4BAIS=y
 CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_SPI_FLASH_SPANSION=y
 CONFIG_SPI_FLASH_STMICRO=y
-CONFIG_PHYLIB=y
-CONFIG_PHYLIB_10G=y
-CONFIG_PHY_AQUANTIA=y
-CONFIG_PHY_BROADCOM=y
-CONFIG_PHY_MARVELL=y
-CONFIG_PHY_MICREL=y
-CONFIG_PHY_MICREL_KSZ9021=y
-CONFIG_PHY_MICREL_KSZ9031=y
-CONFIG_PHY_MSCC=y
-CONFIG_PHY_REALTEK=y
-CONFIG_PHY_VITESSE=y
-CONFIG_NETDEVICES=y
-CONFIG_DM_RTC=y
+CONFIG_PCI=y
+CONFIG_DM_PCI=y
+CONFIG_DM_PCI_COMPAT=y
+CONFIG_PCI_THUNDERX_ECAM=y
+CONFIG_SCSI=y
+CONFIG_DM_SCSI=y
+CONFIG_DM_SERIAL=y
 CONFIG_DEBUG_UART_PL011=y
 CONFIG_DEBUG_UART_BASE=0x87e028000000
 CONFIG_DEBUG_UART_CLOCK=24000000
 CONFIG_DEBUG_UART_SKIP_INIT=y
+CONFIG_PL01X_SERIAL=y
 CONFIG_DM_SPI=y
 CONFIG_CAVIUM_SPI=y
 CONFIG_USB=y
 CONFIG_DM_USB=y
 CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_XHCI_PCI=y
 CONFIG_USB_STORAGE=y
 CONFIG_FAT_WRITE=y
 CONFIG_ERRNO_STR=y
diff --git a/include/configs/thunderx_93xx.h b/include/configs/thunderx_93xx.h
index c7c343a43d..447faaf169 100644
--- a/include/configs/thunderx_93xx.h
+++ b/include/configs/thunderx_93xx.h
@@ -7,62 +7,37 @@
 #ifndef __THUNDERX_93XX_H__
 #define __THUNDERX_93XX_H__
 
+/* Generic Interrupt Controller Definitions */
+#define GICD_BASE			(0x801000000000)
+#define GICR_BASE			(0x801000002000)
 
-/** Thunder 93xx does not support NOR flash */
-#define CONFIG_SYS_NO_FLASH
+/* Generic Timer Definitions */
+#define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
 
 #define CONFIG_SUPPORT_RAW_INITRD
 
 /** Maximum size of image supported for bootm (and bootable FIT images) */
-#define CONFIG_SYS_BOOTM_LEN (256 << 20)
-
-/*
- * Starting uboot and linux kernel from offset 64MB.
- * Below addresses are used for secure ARM-TF memory
- * and RVU mailboxes/MSI-Xes.
- */
-#define MEM_BASE			0x04000000
-
-#define CONFIG_COREID_MASK             0xffffff
-
-/** Low memory base address */
-#define CONFIG_SYS_LOWMEM_BASE		MEM_BASE
-
-/** IPS field of TCR_EL1 */
-#define CONFIG_SYS_TCR_EL1_IPS_BITS	(5UL << 32)
-
-/** IPS field of TCR_EL2 */
-#define CONFIG_SYS_TCR_EL2_IPS_BITS	(5 << 16)
-
-/** IPS field of TCR_EL3 */
-#define CONFIG_SYS_TCR_EL3_IPS_BITS	(5 << 16)
-
-#define CONFIG_SLT
+#define CONFIG_SYS_BOOTM_LEN		(256 << 20)
 
-/* Link Definitions */
-
-/** Code start address */
-#define CONFIG_SYS_TEXT_BASE		0x04000000
+/** Memory base address */
+#define CONFIG_SYS_SDRAM_BASE		CONFIG_SYS_TEXT_BASE
 
 /** Stack starting address */
 #define CONFIG_SYS_INIT_SP_ADDR		(CONFIG_SYS_SDRAM_BASE + 0xffff0)
 
-#define CONFIG_BOARD_LATE_INIT
-
-/* Generic Timer Definitions */
-#define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
-
 /** Memory test starting address */
-#define CONFIG_SYS_MEMTEST_START	MEM_BASE
+#define CONFIG_SYS_MEMTEST_START	CONFIG_SYS_SDRAM_BASE
 
 /** Memory test end address */
-#define CONFIG_SYS_MEMTEST_END		(MEM_BASE + PHYS_SDRAM_1_SIZE)
-
-/* Size of malloc() pool */
+#define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_SDRAM_BASE + 0xf0000)
 
 /** Heap size for U-Boot */
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + 64 * 1024 * 1024)
 
+#define CONFIG_SYS_LOAD_ADDR		CONFIG_SYS_SDRAM_BASE
+
+#define CONFIG_NR_DRAM_BANKS		1
+
 /**
  * This is defined in the board config header.  This specified memory area
  * will get subtracted from the top (end) of memory and won't get
@@ -72,117 +47,33 @@
  */
 #define CONFIG_SYS_MEM_TOP_HIDE		0x1000000
 
-/* Generic Interrupt Controller Definitions */
-
-/** Base address of Global Interrupt Controller GICD registers */
-#define GICD_BASE			(0x801000000000)
-#define GICR_BASE			(0x801000002000)
-
-#define CONFIG_BAUDRATE			115200
-
-/* Net */
-/** Enable ThunderX SMI MDIO driver */
-#define CONFIG_THUNDERX_SMI
-
-/** Enable ThunderX VNIC driver support */
-#define CONFIG_THUNDERX_VNIC
-
-/** Generate a random MAC address if it is not already defined */
-#define CONFIG_RANDOM_MACADDR
-
-#ifndef CONFIG_RANDOM_MACADDR
-/** If no random MAC address, use this address */
-# define CONFIG_ETHADDR			aa:d3:31:40:11:00
-#endif
-
 /**
  * Only allow the Ethernet MAC address environment variable to be
  * overwritten once.
  */
 #define CONFIG_OVERWRITE_ETHADDR_ONCE
 
-/** Command line configuration */
-#define CONFIG_MENU
-
-/*#define CONFIG_MENU_SHOW*/
-/** Enable icache and dcache commands */
-#define CONFIG_CMD_CACHE
-
-#define CONFIG_CMD_DIAG
-
-/** Enable env command */
-#define CONFIG_CMD_ENV
-#undef  CONFIG_CMD_IMLS
-
-#define CONFIG_CMD_MII
-
-/** Enable tftp command */
-#define CONFIG_CMD_TFTP
+/** Reduce hashes printed out */
+#define CONFIG_TFTP_TSIZE
 
-/** Enable pci command */
-#define CONFIG_CMD_PCI
-
-/** Enable askenv command */
-#define CONFIG_CMD_ASKENV
-
-/** Enable environment flags support */
-#define CONFIG_CMD_ENV_FLAGS
-
-/** Enable grepenv command */
-#define CONFIG_CMD_GREPENV
-
-/**
- * Enable callback support when environment variables are set, or
- * changed
- */
-#define CONFIG_CMD_ENV_CALLBACK
-
-#define CONFIG_ENV_CALLBACK_LIST_STATIC "txsmi\\d?mode:smimode"
+/* BOOTP options */
+#define CONFIG_BOOTP_BOOTFILESIZE
+#define CONFIG_BOOTP_BOOTPATH
+#define CONFIG_BOOTP_GATEWAY
+#define CONFIG_BOOTP_HOSTNAME
+#define CONFIG_BOOTP_PXE
+#define CONFIG_BOOTP_TFTP_SERVERIP
 
 /* AHCI support Definitions */
-#define CONFIG_SCSI_AHCI
-
-#ifdef CONFIG_SCSI_AHCI
+#ifdef CONFIG_DM_SCSI
 /** Maximum number of SATA devices per controller*/
-# define CONFIG_SYS_SCSI_MAX_SCSI_ID	1
+#define CONFIG_SYS_SCSI_MAX_SCSI_ID	1
 /** Enable 48-bit SATA addressing */
 # define CONFIG_LBA48
-/** Enable libata, required for SATA */
-# define CONFIG_LIBATA
 /** Enable 64-bit addressing */
 # define CONFIG_SYS_64BIT_LBA
 #endif
 
-#define CONFIG_SYS_SATA_MAX_DEVICE 2
-
-#define CONFIG_SYS_NVME_MAX_DEVICE 64
-
-/* PCIe */
-/** Show devices found on PCI bus */
-#define CONFIG_PCI_SCAN_SHOW
-#undef CONFIG_PCI_ENUM_ONLY
-
-/** Number of I2C buses */
-#define CONFIG_SYS_MAX_I2C_BUS	2
-
-/** I2C slave address */
-#define CONFIG_SYS_I2C_THUNDERX_SLAVE_ADDR	0x77
-
-/** I2C bus real-time clock resides on */
-#define CONFIG_SYS_RTC_BUS_NUM 0
-
-/** I2C address of real-time clcok */
-#define CONFIG_SYS_I2C_RTC_ADDR 0x68
-
-/** Enable Dallas or compabile DS1337 driver */
-#define CONFIG_RTC_DS1337
-
-#define CONFIG_DDR_SPD
-#define CONFIG_SYS_SPD_ADDR_LIST 	\
-	{ 0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57 }
-
-#define CONFIG_SYS_SPD_I2C_BUS 1
-
 /***** SPI Defines *********/
 #ifdef CONFIG_DM_SPI_FLASH
 #define CONFIG_SF_DEFAULT_SPEED 12500000
@@ -190,128 +81,36 @@
 #define CONFIG_SF_DEFAULT_BUS	0
 #define CONFIG_SF_DEFAULT_CS	0
 #endif
-/**************************/
-#define CONFIG_CMD_SAVES
-
-/* BOOTP options */
-#define CONFIG_BOOTP_BOOTFILESIZE
-#define CONFIG_BOOTP_BOOTPATH
-#define CONFIG_BOOTP_GATEWAY
-#define CONFIG_BOOTP_HOSTNAME
-#define CONFIG_BOOTP_PXE
-#define CONFIG_BOOTP_TFTP_SERVERIP
-
-/* Miscellaneous configurable options */
-#define CONFIG_SYS_LOAD_ADDR		(MEM_BASE)
-
-/* Physical Memory Map */
-#define CONFIG_NR_DRAM_BANKS		1
-#define PHYS_SDRAM_1			(MEM_BASE)	  /* SDRAM Bank #1 */
-#define PHYS_SDRAM_1_SIZE		(0x80000000-MEM_BASE)	/* 2048 MB */
-#define CONFIG_SYS_SDRAM_BASE		PHYS_SDRAM_1
-
-/** Enable XHCI PCI driver */
-#define CONFIG_USB_XHCI_PCI
-/** Maximum root ports for XHCI */
-#define CONFIG_SYS_USB_XHCI_MAX_ROOT_PORTS 2
-
-/** Define some USB drivers for ethernet to USB dongles to work */
-#define CONFIG_USB_HOST_ETHER
-#define CONFIG_USB_ETHER_ASIX
-#define CONFIG_USB_ETHER_ASIX88179
-
-/* PCIe network controller drivers */
-/** Intel E1000 network card */
-#define CONFIG_E1000
-#define CONFIG_E1000_SPI
-/** Enable command for E1000 card */
-#define CONFIG_CMD_E1000
-
-/* #define E1000_DEBUG */
-
-/* Initial environment variables */
-#define UBOOT_IMG_HEAD_SIZE		0x40
-/* C80000 - 0x40 */
 
 /** Extra environment settings */
 #define CONFIG_EXTRA_ENV_SETTINGS	\
-					"kernel_addr=0x40080000\0"	\
-					"smi0mode=0.0.0\0"		\
-					"smi1mode=0.0.0\0"		\
+					"kernel_addr=040080000\0"	\
 					"autoload=0\0"
 
-#define CONFIG_BOOTARGS			\
-					"console=ttyAMA0,115200n8 " \
-					"earlycon=pl011,0x87e028000000 " \
-					"debug maxcpus=24 rootwait rw "\
-					"root=/dev/sda2 coherent_pool=16M"
-
-/** Store U-Boot version in "ver" environment variable */
-#define CONFIG_VERSION_VARIABLE
-
-/** Where the environment is located */
-
-/* Use ATF based calls for env variables */
-/*#define CONFIG_ATF*/
-
-/* Use uboot SPI APIs for env variables */
-#define CONFIG_SPI_ENV
-
-/* Use early board init for pci_init call */
-#define CONFIG_BOARD_EARLY_INIT_R
-
-#if defined(CONFIG_ATF)
-#define CONFIG_ENV_IS_IN_ATF
-#define CONFIG_SYS_ENV_ATF_NOR
-#elif defined(CONFIG_SPI_ENV)
-#define CONFIG_ENV_IS_IN_SPI_FLASH
-#define CONFIG_ENV_SECT_SIZE		(64 * 1024)
+/** Environment defines */
+#define CONFIG_ENV_SIZE			0x8000
+#define CONFIG_ENV_OFFSET		0xf00000
+#if defined(CONFIG_ENV_IS_IN_MMC)
+#define CONFIG_SYS_MMC_ENV_DEV		0
+#elif defined(CONFIG_ENV_IS_IN_SPI_FLASH)
+#define CONFIG_ENV_SECT_SIZE		(64*1024)
 #define CONFIG_ENV_SPI_MAX_HZ		12500000
 #define CONFIG_ENV_SPI_MODE		0
 #define CONFIG_ENV_SPI_BUS		0
 #define CONFIG_ENV_SPI_CS		0
 #endif
 
-/** Size of environment in bytes */
-#define CONFIG_ENV_SIZE			0x8000
-/** Starting offset of the environment */
-#define CONFIG_ENV_OFFSET		0xf00000
-
-/** Maximum size of the BDK flat device-tree */
-#define CONFIG_BDK_FDT_SIZE		0x20000
-
-#define CONFIG_THUNDERX_PROMPT_SIZE	32
-
 /* Monitor Command Prompt */
 #define CONFIG_SYS_CBSIZE		1024	/** Console I/O Buffer Size */
-#define CONFIG_SYS_PBSIZE		(CONFIG_SYS_CBSIZE + \
-					 CONFIG_THUNDERX_PROMPT_SIZE + 16)\
-
-#define CONFIG_SYS_PROMPT_HUSH_PS2	"> "
 #define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
 
-#undef CONFIG_SYS_PROMPT
-#define CONFIG_SYS_PROMPT		getenv("prompt")
-
-/** Enable long help support */
-#define CONFIG_SYS_LONGHELP
-/** Enable editing of the command line */
-#define CONFIG_CMDLINE_EDITING
-/** Enable tab autocomplete on command line */
-#define CONFIG_AUTO_COMPLETE
-#define CONFIG_FEATURE_COMMAND_EDITING
-#define CONFIG_SYS_MAXARGS		64		/** max command args */
-#define CONFIG_NO_RELOCATION		1
-
-/** System PLL reference clock */
-#define PLL_REF_CLK			50000000	/* 50 MHz */
+#define CONFIG_SYS_MAXARGS		64	/** max command args */
 
-#define NS_PER_REF_CLK_TICK		(1000000000/PLL_REF_CLK)
-
-#define CONFIG_OCTEON_MMC
 #define CONFIG_SYS_MMC_MAX_BLK_COUNT	8191
-#define CONFIG_SYS_MMC_SET_DEV
 
-#define CONFIG_CMD_TIME
+#undef CONFIG_SYS_PROMPT
+#define CONFIG_SYS_PROMPT		env_get("prompt")
+
+#define CONFIG_HW_WATCHDOG
 
 #endif /* __THUNDERX_93XX_H__ */
-- 
2.29.0

